data_sender.py

```diff
@@ -25,20 +26,32 @@ def get_power_port_data(self):
 
         return int(100 * dx), int(100 * dy), int(10 * round(angle, 1))
 
+    def get_ball_detection_data(self):
+        ball_data, obstacle_present = self.ball_finder.get_balls()
+
+        formatted = []
+        for ball in ball_data:
+            formatted.append([9999, 9999] if not ball else [int(100 * ball[0]), int(10 * round(ball[1], 1))])
+
+        return formatted, obstacle_present
```

Comment:
I think it makes more sense to return the string you're going to append to the data being sent, rather than packing and unpacking it in tuples. In general, try to return a single thing as much as possible.